tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h

```diff
@@ -102,8 +109,26 @@ struct OptimizationProfileConfig {
     }
     return true;
   }
+
+ private:
+  // In dynamic shape mode the optimization profile dims are only allowed to
+  // differ from the network input dims where the network input dims have -1
+  // values. We ensure that prof_dim matches the network input (mask).
+  void applyMask(nvinfer1::Dims* prof_dim, const nvinfer1::Dims& mask) {
+    for (int i = 0; i < mask.nbDims; i++) {
+      if (mask.d[i] != -1) {
+        prof_dim->d[i] = mask.d[i];
+      }
+    }
+  }
 };
 
+// Optimization profile generation strategies.
+//
+// The kUnitTest option is only used for unit testing to avoid TRT bug that
+// removes redundant input shape values (200641519).
+enum class ProfileStrategy { kImplicitBatchLike, kMultiple, kUnitTest };
```

Comment:
How about doing all these:
  (1) remove kUnitTest
  (2) implement kImplicitBatchLike the way you implement kUnitTest now
  (3) add the comment to the place where we increment the batch size by one to workaround the bug

Reply:
I have the following two objections:
- The user might have input shape [N,C,H,W] defined in a way that the network barely fits into the memory. If we change the profile to [N+1,C,H,W] we risk running out of the memory for those cases.
- By changing max, we are affecting which kernels could be selected by TRT, and that could reduce performance in certain cases. 
Because of this I am strongly against modifying the Profile strategies that are exposed to the user.

Reply:
But does the bug we mentioned here essentially means kImplicitBatchLike won't work until the bug is fix?